tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,203 @@
+                    # hints that work in Pandas
+                    addl_hints = {
+                        'datetimeformat': f'{dateformat} HH24:MI:SS',
+                        'datetimeformattz': f'{dateformat} HH:MI:SS',
+                    }
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'dateformat': dateformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            csv_text = self.unload(column_name='date',
+                                   records_format=records_format)
+            self.assertEqual(csv_text, create_sample(dateformat) + "\n",
+                             f"from dateformat {dateformat} and addl_hints {addl_hints}")
+
+    def test_unload_datetime(self) -> None:
+        self.createDateTimeTable()
+        matching_dateformat = {
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH12:MI AM': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+        }
+        for datetimeformat in DATETIME_CASES:
+            addl_hints: PartialRecordsHints = {}
+            if self.engine.name == 'redshift':
+                if datetimeformat != 'YYYY-MM-DD HH24:MI:SS':
+                    # this is the only format supported by Redshift on
+                    # export, so we're going to need to be sure to use
+                    # hints that work in Pandas - i.e.,
+                    addl_hints = {
+                        'dateformat': matching_dateformat[datetimeformat],
+                        'datetimeformattz': datetimeformat,
+                    }
+                    if 'AM' in datetimeformat:
+                        # TODO: Add a GitHub issue for this
+                        logger.warning('Cannot export this dateformat using Pandas or Redshift--'
+                                       'skipping test')
+                        continue
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformat': datetimeformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            csv_text = self.unload(column_name='timestamp',
+                                   records_format=records_format)
+            self.assertIn(csv_text, [create_sample(datetimeformat) + "\n",
+                                     # TODO: Should this be necessary?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker